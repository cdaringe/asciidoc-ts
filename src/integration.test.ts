import { describe, expect, test } from "vitest";
import { toAST } from "./mod.js";

test("Complex Document", async ({ expect }) => {
  const input = `
= Main Header

== Section 1

This is a paragraph with *bold* and _italic_ text.

* List item 1
* List item 2
** Nested item

[source,javascript]
----
function greet(name) {
console.log(\`Hello, ${"\${name}"}!\`);
}
----

https://example.com[Visit Example]

NOTE: This is an admonition.
`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});
