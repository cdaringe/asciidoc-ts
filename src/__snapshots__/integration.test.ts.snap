// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Complex Document 1`] = `
{
  "blocks": [
    {
      "type": "BlankLine",
    },
    {
      "content": [
        {
          "content": "Main Header",
          "type": "PlainText",
        },
      ],
      "level": 1,
      "type": "Header",
    },
    {
      "content": [
        {
          "content": "Section 1",
          "type": "PlainText",
        },
      ],
      "level": 2,
      "type": "Header",
    },
    {
      "content": [
        {
          "content": "This is a paragraph with ",
          "type": "PlainText",
        },
        {
          "content": [
            {
              "content": "bold",
              "type": "PlainText",
            },
          ],
          "type": "ConstrainedBold",
        },
        {
          "content": "and ",
          "type": "PlainText",
        },
        {
          "content": [
            {
              "content": "italic",
              "type": "PlainText",
            },
          ],
          "type": "UnconstrainedItalic",
        },
        {
          "content": "text.",
          "type": "PlainText",
        },
      ],
      "type": "Paragraph",
    },
    {
      "items": [
        {
          "content": [
            {
              "content": "List item 1",
              "type": "PlainText",
            },
          ],
          "depth": 0,
          "type": "UnorderedListItem",
        },
        {
          "content": [
            {
              "content": "List item 2",
              "type": "PlainText",
            },
          ],
          "depth": 0,
          "type": "UnorderedListItem",
        },
        {
          "content": [
            {
              "content": "Nested item",
              "type": "PlainText",
            },
          ],
          "depth": 1,
          "type": "UnorderedListItem",
        },
      ],
      "type": "UnorderedList",
    },
    {
      "content": "function greet(name) {
console.log(\`Hello, \${name}!\`);
}",
      "delimited": true,
      "metadata": {
        "attributes": [
          {
            "name": "source",
            "type": "AttributeEntry",
          },
          {
            "name": "javascript",
            "type": "AttributeEntry",
          },
        ],
      },
      "type": "CodeBlock",
    },
    {
      "content": [
        {
          "attributes": [
            {
              "name": "Visit Example",
              "type": "AttributeEntry",
            },
          ],
          "scheme": "https",
          "type": "UrlMacro",
          "url": "https://example.com",
        },
      ],
      "type": "Paragraph",
    },
    {
      "admonitionType": "NOTE",
      "content": "This is an admonition.",
      "type": "Admonition",
    },
  ],
  "type": "Document",
}
`;
